### ISSUE TITLE
Feature request: alt_text on tf.summary name

### ISSUE THREAD

**Issue by ataiya** [tensorflow/tensorboard#1920]
Most people rely on the log_dir name to identify an experiment, but this can get out of hand pretty quickly.

It would be amazing if there was an easily reachable section in TensorBoard where I could see more details for a particular run (some people use "Text" summaries, but these are not easily visible from any tab). 

A simple way would be to use an alt_text (the text box that appears when the mouse is stationary over an item).

This is how the TensorBoard API could be adapted to support this:
```
logdir='n_samples_1/20170530_141605'
description='more details on experiment can be provided here, and visualized like this via an html "alt_text"'
writer = tf.contrib.summary.create_file_writer(log_dir, description=description)
```

This is what it would look like:
![image](https://user-images.githubusercontent.com/4480573/53582319-3e954000-3b4d-11e9-802e-4a407676316e.png)

**Comment by killolgovani** [comment]
@nfelt Can I work on this issue?

**Comment by nfelt** [comment]
@killolgovani Thanks for your interest.  I think this feature request needs more design work before we can decide both whether to add it and how.  In particular, it would be helpful to get a sense of what kind of information is useful to include in a tooltip for the run - if it's structured information like hyperparameters there might be a better way to communicate this rather than open-ended text.  @ataiya was there a specific use case you had in mind?

Along those lines, this feature request also seems to overlap with #210 and #194 which are both about providing runs with more structured information like tags or metadata, which could be used for searching in addition to just displaying a tooltip.

Finally, common to all of these feature requests is a need to figure out how to store the data in the events file and how to dedupe multiple instances of it within a given run.  #210 also is interested in being able to tag/label runs after the fact, in which case we might want a way to store this information that can be persisted, which introduces a whole separate set of questions.

**Comment by ataiya** [comment]
1) The example I gave was for hyper-parameters, but it could be anything the user wants to put there. E.g. a textual description of what one was trying in the experiment for which a direct association to control-flow in the code is missing (e.g. toggling a chunk of code and re-running the training session). The core limitation is that people like to keep the "logdir" name short, yet simultaneously you currently want the name to give you as much information about the experiment as possible. This would resolve this trade-off.

2) This seems orthogonal to *just* show hyperparameters? Further, note the type of tooltip could be formatted according to the type that is passed to it dynamically. e.g. a raw string would be formatted as classical tooltip, while if you pass an hparam instance you could show, e.g., a table.
But honestly I would start simple with a string, and leave the door open to extensions.

3) Indeed, that seems to need some thinking. Perhaps you can reuse global step "-1" or "None" to store info that is persistent through an entire training session? Just a random thought.

**Comment by killolgovani** [comment]
@nfelt Okay, thank you for the insights, looking forward to making contributions in the future.
